tests/unit/query_engines/test_mssql_dialect.py

```diff
@@ -33,10 +34,34 @@ def test_mssql_date_types():
         _str(datetime_col == 2021)
 
 
+def test_select_star_into():
+    table = sqlalchemy.table("foo", sqlalchemy.Column("bar"))
+    query = sqlalchemy.select(table.c.bar).where(table.c.bar > 1)
+    target_table = sqlalchemy.table("test")
+    select_into = SelectStarInto(target_table, query.alias())
+    assert _str(select_into) == (
+        "SELECT * INTO test FROM (SELECT foo.bar AS bar \n"
+        "FROM foo \n"
+        "WHERE foo.bar > 1) AS anon_1"
+    )
+
+
+def test_select_star_into_can_be_iterated():
+    table = sqlalchemy.table("foo", sqlalchemy.Column("bar"))
+    query = sqlalchemy.select(table.c.bar).where(table.c.bar > 1)
+    target_table = sqlalchemy.table("test")
+    select_into = SelectStarInto(target_table, query.alias())
+
+    # Check that SelectStarInto supports iteration by confirming that we can get back to
+    # the original column element if we iterate it
+    elements = iterate(select_into)
+    matches = [el for el in elements if el is table.c.bar]
+    assert len(matches) > 0
```

Comment:
I don't really understand this.  How would it fail?

Reply:
If you don't implement the `get_children()` method (as I didn't originally) then you won't get any errors from `iterate()`, it will just fail to recurse into the elements children.

I can (and will) make this clearer in the test comments.